### ISSUE TITLE
Cleanup and update `releng/test_config.yaml`

### ISSUE THREAD

**Issue by xmudrii** [kubernetes/test-infra#32340]
[`releng/test_config.yaml`](https://github.com/kubernetes/test-infra/blob/master/releng/test_config.yaml) contains definitions for `kubernetes/kubernetes` tests (mainly periodics). We use those definitions to generate jobs for release branches when cutting a new release branch.

Over the time, this file has become unmaintained which is a huge risk for having jobs that are not **properly** testing release branches.

Here are some problems that I observed; this is a non exhaustive list, in other words, there might be other issues that I didn't observe:

- [ ] `.jobs`
  - [ ] `ci-kubernetes-e2e-gce-cos-.+?-default` is inconsistent
    - [ ] Different release branches has different args (to confirm if intentional)
      - [ ] Only `ci-kubernetes-e2e-gce-cos-k8sbeta-default` has `--env=ENABLE_CACHE_MUTATION_DETECTOR=true` arg
      - [ ] Only `ci-kubernetes-e2e-gce-cos-k8sstable1-default` has `--env=ENABLE_POD_SECURITY_POLICY=true`
      - [ ] Other `ci-kubernetes-e2e-gce-cos-.+?-default` jobs have no (additional args)
    - [ ] `k8sbeta` and `k8sstable1` jobs are missing `testgridNumFailuresToAlert` parameter
  - [ ] We're missing `betaapis` tests for `k8sbeta`, `k8sstable1`, and `k8sstable2` releases
    - [ ] Verify if args for `betaapis` tests are still valid
- [ ] `.images`
  - [ ] `cos1`, `cos2`, `ubuntu1`, and `ubuntu2` images are not present in jobs names and therefore are likely unused (check if it is safe to remove these images)
- [ ] `.testSuites`
  - [ ] `flaky`, `soak`, `stackdriver`, `updown`, and `nosat` suites are not present in jobs names and therefore are likely unused (check if it is safe to remove these images)
- [ ] `.nodeImages`
  - [ ] `cos1`, `cos2`, `ubuntu1`, and `ubuntu2` images are not present in jobs names and therefore are likely unused (check if it is safe to remove these images)
- [ ] `.nodeTestSuites`
  - [ ] `gkespec` and `serial` are likely unused

Once these issues are addressed, we should verify if arguments are up to date for remaining test options.

cc @kubernetes/release-engineering @dims

**Comment by xmudrii** [comment]
We're running into this issue again ahead of the 1.31 jobs cut. These are the high priority issues at the moment:

- [`ci-kubernetes-e2e-gce-cos-k8sbeta-default` (1.30)](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L48-L53) has `--env=ENABLE_CACHE_MUTATION_DETECTOR=true`, do we need to keep this for 1.31 as well?
- [`ci-kubernetes-e2e-gce-cos-k8sbeta-alphafeatures` (1.30)](https://github.com/kubernetes/test-infra/blob/master/releng/test_config.yaml#L64C3-L68) uses a custom testSuite called [`alphafeatures-ccm-eventedpleg`](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L305-L318), do we need to keep it for 1.31 as well?
- [`ci-kubernetes-e2e-gce-cos-k8sstable1-default` (1.29)](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L79-L85) has `--env=ENABLE_POD_SECURITY_POLICY=true`, however, this env is present only for 1.29. Do we need it for any other version or is it specific to only 1.29?
- We have `betaapis` tests only for [1.26](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L202-L211) and [1.27](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L160-L169). Are we getting rid of the `betaapis` tests or do we need those tests for all other release branches?

/priority critical-urgent
cc @dims @mehabhalodiya

**Comment by dims** [comment]
> [ci-kubernetes-e2e-gce-cos-k8sbeta-default (1.30)](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L48-L53) has --env=ENABLE_CACHE_MUTATION_DETECTOR=true, do we need to keep this for 1.31 as well?

Yes, we should keep it. helps catch the scenario where `anything mutates a shared informer cache` (see description [here](https://github.com/kubernetes-sigs/kubetest2/blob/53f3d216ad9fba694bc27c13c9ebf4dfb37b73d9/kubetest2-gce/deployer/deployer.go#L87))

> [ci-kubernetes-e2e-gce-cos-k8sbeta-alphafeatures (1.30)](https://github.com/kubernetes/test-infra/blob/master/releng/test_config.yaml#L64C3-L68) uses a custom testSuite called [alphafeatures-ccm-eventedpleg](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L305-L318), do we need to keep it for 1.31 as well?

Yes, please. will help give us signal on an alpha feature https://github.com/kubernetes/kubernetes/blob/9413cf204ac92711cc8aff472b1ed11ba79760ac/pkg/features/kube_features.go#L261-L267 

> [ci-kubernetes-e2e-gce-cos-k8sstable1-default (1.29)](https://github.com/kubernetes/test-infra/blob/76fdd1911197d157e59e966363b4529cdae897fd/releng/test_config.yaml#L79-L85) has --env=ENABLE_POD_SECURITY_POLICY=true, however, this env is present only for 1.29. Do we need it for any other version or is it specific to only 1.29?

We can drop it, does not do much AFAICT looking through references - https://cs.k8s.io/?q=ENABLE_POD_SECURITY_POLICY%7CEnablePodSecurityPolicy&i=nope&files=&excludeFiles=vendor&repos=

**Comment by BenTheElder** [comment]
Can we please just use config fork annotations on normal prowjobs and stop using this other thing entirely?

I don't think SIG release should be defining these jobs, just forking them when new branches are cut, and I really doubt anyone else in the project is looking at this file. It's hard to even understand what these config fields are doing.

cc @liggitt @tallclair re: PSP

yes we want to keep the cache mutation detector enabled.

**Comment by liggitt** [comment]
ENABLE_POD_SECURITY_POLICY is irrelevant in 1.25+ and can be dropped

**Comment by BenTheElder** [comment]
documented and widely used:

https://github.com/kubernetes/test-infra/tree/master/releng/config-forker#supported-annotations

Versus ~no docs for generate_tests and the out of band config.

If the config fork annotations are not powerful enough for some reason, let's fix that.

**Comment by xmudrii** [comment]
> Can we please just use config fork annotations on normal prowjobs and stop using this other thing entirely?

Long term, we can look into it. But short term and specially for this upcoming release, definitely no. We're happy to look into it, but this will require some time and we want to get the release branch up and running ASAP.

**Comment by xmudrii** [comment]
/assign
Let met look into this and if we can quickly drop this

**Comment by Rajalakshmi-Girish** [comment]
> Can we please just use config fork annotations on normal prowjobs and stop using this other thing entirely?
> 
> I don't think SIG release should be defining these jobs, just forking them when new branches are cut, and I really doubt anyone else in the project is looking at this file. It's hard to even understand what these config fields are doing.

@BenTheElder @xmudrii Can you please clarify if my understanding is correct?

The [generated](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/generated/generated.yaml) release branch jobs  which are currently being generated by tool [generate-jobs](https://github.com/kubernetes/test-infra/tree/master/releng#generate-jobs) must be instead created by tools [config-forker](https://github.com/kubernetes/test-infra/blob/master/releng/config-forker/README.md) and [config-rotator](https://github.com/kubernetes/test-infra/tree/master/releng/config-rotator#config-rotator).

Can this be achieved by following below?

1. Create a new file may be, by name `cloud_provider_image_validation.yaml` in [releng](https://github.com/kubernetes/test-infra/tree/master/releng) folder which will contain provider/image validation jobs in detail using https://github.com/kubernetes/test-infra/blob/master/releng/test_config.yaml as reference (As of now there are six jobs for each version_marker )

2. Use `config-forker` and generate k8sbeta release yaml at a new folder `cloud-provider` in https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-release/release-branch-jobs

3. Then use `config-rotator` and rotate the k8sbeta job to stable1, stable2, stable3 (May need some manual arg changes as per release version)

4. https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/generated folder will go away.
5. Getting rid of [test_config.yaml](https://github.com/kubernetes/test-infra/blob/master/releng/test_config.yaml) and [generate_tests.py](https://github.com/kubernetes/test-infra/blob/master/releng/generate_tests.py) will be next steps.

**Comment by xmudrii** [comment]
@Rajalakshmi-Girish is working on this at the moment. Please let me know if you need anything!
/unassign
/assign @Rajalakshmi-Girish

**Comment by BenTheElder** [comment]
These jobs also wind up using the master branch image on all branches.

I think it would be best to just migrate them to normal jobs that get forked, I'm not sure why we did this.

**Comment by saschagrunert** [comment]
Should be done, the file is gone. 

/close
